xds/src/main/java/io/grpc/xds/internal/sds/XdsServer.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.internal.sds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.grpc.Server;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.xds.XdsClientWrapperForServerSds;
+
+import java.io.IOException;
+import java.net.SocketAddress;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Wraps a {@link Server} delegate and {@link XdsClientWrapperForServerSds} and intercepts
+ * {@link Server#shutdown()} etc to properly shut down the {@link XdsClientWrapperForServerSds}
+ * object.
+ */
+final class XdsServer extends Server {
+  private final Server delegate;
+  private final XdsClientWrapperForServerSds xdsClientWrapperForServerSds;
+
+  static Server newInstance(
+      Server delegate, XdsClientWrapperForServerSds xdsClientWrapperForServerSds) {
```

Comment:
add @Nullable for 2nd param. overall methods returns null should also be marked with the annotation. can you check other methods/params/instance variables for the classes in this PR?